providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/UnleashProvider.java

```diff
@@ -0,0 +1,171 @@
+        UnleashConfig unleashConfig = unleashOptions.getUnleashConfigBuilder().build();
+        unleash = new DefaultUnleash(unleashConfig);
+
+        // else, state will be changed via UnleashSubscriberWrapper events
+        if (unleashConfig.isSynchronousFetchOnInitialisation()) {
+            state = ProviderState.READY;
+        } else {
+            log.info("ready state will be changed via UnleashSubscriberWrapper events");
+        }
+
+        log.info("finished initializing provider, state: {}", state);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public void emitProviderReady(ProviderEventDetails details) {
+        super.emitProviderReady(details);
+        state = ProviderState.READY;
+    }
+
+    @Override
+    public void emitProviderError(ProviderEventDetails details) {
+        super.emitProviderError(details);
+        state = ProviderState.ERROR;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        boolean featureBooleanValue = unleash.isEnabled(key, context, defaultValue);
+        return ProviderEvaluation.<Boolean>builder()
+            .value(featureBooleanValue)
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        Variant evaluatedVariant = unleash.getVariant(key, context);
+        String variantName;
+        String value;
+        if (DISABLED_VARIANT.equals(evaluatedVariant)) {
+            variantName = null;
+            value = defaultValue;
+        } else {
+            variantName = evaluatedVariant.getName();
+            value = evaluatedVariant.getPayload().get().getValue();
+        }
+        return ProviderEvaluation.<String>builder()
+            .value(value)
+            .variant(variantName)
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
```

Comment:
Payload of type "number" will come soon in Unleash.

Reply:
Ok, so this can be updated when a new Unleash version is released and adopted.

Reply:
It's already on main, and on the demo instance.
![image](https://github.com/open-feature/java-sdk-contrib/assets/158948/35380211-dfc7-4147-8585-dfa96ee53ea5)

Reply:
Bug...
![image](https://github.com/open-feature/java-sdk-contrib/assets/77168114/3984341d-36c5-47a4-896c-4784f2b5a042)

Reply:
Added number support

Reply:
Hey @liran2000, that bug looks super ugly, The dev who worked on this is off today, so I'll check tomorrow but we can't reproduce this. Was this still an issue last you checked?

Reply:
Not reproduced for me anymore.